torch/csrc/jit/script/compiler.cpp

```diff
@@ -1668,11 +1704,18 @@ struct to_ir {
       // if it's a list, emit a regular index selection op
       return emitBuiltinCall(
                  loc, *graph, aten::select, {gatherable, idx}, {}, true);
-
     } else {
       JIT_ASSERT(gatherable.value(*graph)->type()->isSubtypeOf(DynamicType::get()));
-      return emitSelect(
-          loc, gatherable.value(*graph), /*dim=*/0, idx.value(*graph));
+      auto * index = idx.value(*graph);
+      const auto index_type = index->type();
+      if (index_type->isSubtypeOf(DynamicType::get())) {
+        return emitIndex(loc, gatherable.value(*graph), {index});
+      } else if (index_type == IntType::get()) {
+        return emitSelect(
+            loc, gatherable.value(*graph), /*dim=*/0, index);
+      }
+      throw ErrorReport(loc)
+        << "Cannot index tensor with value of type " << index_type->str();
```

Comment:
It's really annoying that we have this thing implemented in multiple places. Can't we just call `emitMultidimIndexing` in here?

Reply:
Can do. The code is a little annoying to work around because emitMultidimIndex assumes that only the `gatherable` has been emitted, but `emitIndex` emits both the `gatherable` and the `index` and I would have to tell `emitIndex` to only emit the `gatherable`. I'll do some refactoring.